docs/source/create/materials/procedural-shaders.rst

```diff
@@ -0,0 +1,443 @@
+Shaders are applied to entities and avatars by way of attaching them to a material. That material is then attached to your entity or avatar, therefore applying the shader as intended.
+
+If you are unfamiliar with material entities, you can find more information `here <https://docs.projectathena.dev/create/entities/material-entity.html>`_.
+
+Material entities have data that is stored in the JSON format, and they have a property that is called ``materialData``. The ``materialData`` property will require model and procedural fields. Here is an example::
+
+    {
+        "materials": [{
+            "model": "hifi_shader_simple"
+            "procedural": {
+            "version": 3,
+            "shaderUrl": "https://docs.projectathena.dev/_static/resources/Proceduralv3.fs"
+            }
+        }]
+    }
+
+The ``materialData`` JSON can be applied either via the Project Athena Interface’s edit tools or with a script. Here's another example::
+
+    Entities.addEntity({
+    	type: "Material",
+    	materialURL: "materialData",
+    	priority: 1,
+    	materialData: JSON.stringify({
+    		materials: {
+    			"model": "hifi_shader_simple"
+    			"procedural": {
+    			  	"version": 3,
+    			  	"shaderUrl": "https://docs.projectathena.dev/_static/resources/Proceduralv3.fs"
+    			}
+    		}
+    	})
+    });
+
+You must specify the material "model" as ``hifi_shader_simple`` and provide a shader link. To provide a fragment shader, set ``fragmentShaderURL`` (or ``shaderUrl``). To provide a vertex shader, set ``vertexShaderURL``.
+
+^^^^^^^^^^^^^^^
+Shader Template
+^^^^^^^^^^^^^^^
+
+When you learn about shaders for other applications, the shader may have a function like ``main()`` that is run first. By contrast, Project Athena has a specific function name that must be called. Which function is used depends on which version of the shader you use.
+
+As shaders were developed, features for them evolved a bit over time. As a result, there are several shader versions, and each version has a different call signature. **Versions 1 and 2** are the oldest, and will still work. **Versions 3 and 4** are the newest and expose more features. Version 4 provides for per-fragment positions, however it is also the most expensive. Therefore it is recommended to use Version 3 if that extra feature from Version 4 is not needed.
+
+A basic template for a shader will look something like this example::
+
+    // version 3
+    float getProceduralFragment(inout ProceduralFragment data) {
+        data.diffuse = vec3(0);
+        data.occlusion = 0;
+        data.roughness = 1;
+        data.emissive = _positionMS.xyz;
+        return 0; // "emissiveAmount", either <=0 or >0, suggest return 0 and use data.emissive
+    }
+
+The function ``getProceduralFragmentWithPosition()`` is the default main entry point for the fragment shader. Because shaders are always read by their compiler from top to bottom, this function must always be the last one in your shader code.
```

Comment:
most of this paragraph and the next are already covered below (also they refer to v4 things but the example is v3)

Reply:
https://docs.projectathena.dev/staging/html/create/materials/procedural-shaders.html#shader-template Here is the revision.